openfl/interface/interactive_api/experiment.py

```diff
@@ -252,6 +264,21 @@ def _prepare_plan(self, model_provider, task_keeper, data_loader,
         # We also could change the aggregator logic so it will send tasks to aggregator
         # as soon as it connects. This change should be a part of a bigger PR
         # brining in fault tolerance changes
+
+        # Check tasks type
+        for name in task_keeper.task_registry:
+            if task_keeper.task_contract[name]['optimizer'] is not None:
+                self.train_task_exist = True
+            else:
+                self.validation_task_exist = True
+
+        if not self.train_task_exist:
+            # Since we have only validation task, we do not have to train it multiple times
+            if rounds_to_train != 1:
```

Comment:
It may be better to return an error than to implicitly change user parameters.

Reply:
It changes in explicitly way, there is some logged info:
```Python3
            self.logger.info('Variable rounds_to_train was changed to 1, '
                             'because only validation task(s) were given')
```